file/file.go

```diff
@@ -140,29 +141,37 @@ func (s *fileSub) Close() error {
 type SubTopic string
 
 // Subscribe subscribes/consumes records from file.
-func (t SubTopic) Subscribe(ctx context.Context, handler bps.Handler, _ ...bps.SubOption) error {
-	f, err := os.Open(string(t))
-	if os.IsNotExist(err) {
-		return nil
-	} else if err != nil {
-		return err
-	}
-	defer f.Close()
-
-	dec := json.NewDecoder(f)
-	for dec.More() {
-		if err := ctx.Err(); err != nil {
+func (t SubTopic) Subscribe(handler bps.Handler, _ ...bps.SubOption) (bps.Subscription, error) {
+	sub := concurrent.NewGroup(context.Background())
```

Comment:
pass a std error handler to `concurrent.NewGroup`, instead of using an errgroup, just run a `for` loop
```
sub := concurrent.NewGroup(context.Background(), func(err error) { log.Printf("[bps] error on subscription to %q: %v", t, err) )
```

Reply:
or, even pass one to the invocation itself:

```
sub := concurrent.NewGroup(context.Background())
sub.Go(func() error { ...  }, func(err error) { ... })
```

Reply:
I'll drop the `os.IsNotExist` "success" behaviour then.

Reply:
And instead of passing error handler to `concurrent.Group`, I'll just make `.Go(func())` (error-less)

Reply:
And will leave error handling for next PR.